### ISSUE TITLE
Try to use WAL mode for SQLite

### ISSUE THREAD

**Issue by miso-belica** [honzajavorek/junior.guru#383]
Today I read the article https://honzajavorek.cz/blog/tydenni-poznamky-novy-seznam-nabidek-prace/#retries and was wondering if you are using [WAL mode in SQLite](https://sqlite.org/wal.html) because fo the sentence _Řešit to nejde, SQLite na intenzivní zápis není dělaná_. I know the SQLite has its limits but it can be pretty powerful with WAL. You can read more in https://dev.to/victoria/sqlite-in-production-with-wal-3387 or other articles on the internet.

I can see even [Peewee has convenient support](http://docs.peewee-orm.com/en/latest/peewee/database.html?highlight=journal_mode#using-sqlite). So instead of this 
https://github.com/honzajavorek/junior.guru/blob/a1d0926f4e27781d9d1de13eeff4ad8c1f3bedb5/juniorguru/models/base.py#L17 (and maybe https://github.com/honzajavorek/junior.guru/blob/master/scripts/check_scrapers.py#L8) you can try to use.

```python
db = SqliteDatabase(db_file, check_same_thread=False, pragmas={'journal_mode': 'wal'})
```

Let me know your findings, I am curious :)

**Comment by honzajavorek** [comment]
Hi, thanks for the suggestion! Just reading the linked article. It's super helpful for understanding the difference. I think I've noticed WAL somewhere before, but I didn't realized it could help me as an easy optimization. The implementation looks very convenient, too. I'll try it out and let's see!

I don't have running stats of how many retries per run I have currently, so the outcome won't be measurable. I'll try to report back though if I'll feel that subjectively the number of retries has improved or the time needed for one run has dropped.

**Comment by honzajavorek** [comment]
The article explains it very well and it's clear now WAL should be the way to go for me if I have concurrent and write-heavy process like this. Cool 🚀

**Comment by honzajavorek** [comment]
Implemented in ed84c30
